cartography/intel/gcp/dns.py

```diff
@@ -121,6 +121,103 @@ def get_dns_rrs(dns: Resource, dns_zones: List[Dict], project_id: str) -> List[R
             # raise
             return []
 
+@timeit
+def get_dns_keys(dns: Resource, dns_zones: List[Dict], project_id: str) -> List[Resource]:
+    """
+    Returns a list of DNS Keys within the given project.
+
+    :type dns: The GCP DNS resource object
+    :param dns: The DNS resource object created by googleapiclient.discovery.build()
+
+    :type dns_zones: list
+    :param dns_zones: List of DNS zones for the project
+
+    :type project_id: str
+    :param project_id: Current Google Project Id
+
+    :rtype: list
+    :return: List of Resource Record Sets
+    """
+    try:
+        dns_keys = []
+        for zone in dns_zones:
+            request = dns.dnsKeys().list(project=project_id, managedZone=zone['name'])
+            while request is not None:
+                response = request.execute()
+                if response.get('dnsKeys'):
+                    for key in response['dnsKeys']:
+                        key['zone'] = zone['id']
+                        key['id'] = f"projects/{project_id}/managedZones/{zone['name']}/dnsKeys/{key['id']}"
+                        dns_keys.append(key)
+                request = dns.dnsKeys().list_next(previous_request=request, previous_response=response)
+        return dns_keys
+    except HttpError as e:
+        err = json.loads(e.content.decode('utf-8'))['error']
+        if err.get('status', '') == 'PERMISSION_DENIED' or err.get('message', '') == 'Forbidden':
+            logger.warning(
+                (
+                    "Could not retrieve DNS Keys on project %s due to permissions issues. Code: %s, Message: %s"
+                ), project_id, err['code'], err['message'],
+            )
+            return []
+        else:
+            # raise
+            return []
+
+@timeit
+def get_dns_policies(dns: Resource, project_id: str, common_job_parameters) -> List[Resource]:
+    """
+    Returns a list of DNS policies within the given project.
+
+    :type dns: The GCP DNS resource object
+    :param dns: The DNS resource object created by googleapiclient.discovery.build()
+
+    :type project_id: str
+    :param project_id: Current Google Project Id
+
+    :rtype: list
+    :return: List of DNS Policies
+    """
+    try:
+        policies = []
+        request = dns.policies().list(project=project_id)
+        while request is not None:
+            response = request.execute()
+            if response.get('policies'):
+                for policy in response['policies']:
```

Comment:
same as above. let's create a transform method and move the logic to populate additional fields into that method.

Reply:
Like for this policy we are populating only the `id`. The transform method will be very small